server/src/main/java/io/crate/analyze/validator/SemanticSortValidator.java

```diff
@@ -68,16 +68,16 @@ public static void validate(Symbol symbol) throws UnsupportedOperationException
      * @throws UnsupportedOperationException
      */
     public static void validate(Symbol symbol, String operation) throws UnsupportedOperationException {
-        symbol.accept(INNER_VALIDATOR, new SortContext(operation));
+        symbol.accept(INNER_VALIDATOR, new SortContext(operation, symbol instanceof Function));
```

Comment:
I think we don't need this.

What I had in mind was only removing `context.inFunction = false.` 

If we are checking ORDER BY and see a Function `some_func1` it means we do `ORDER BY some_func2`. 

We can have standalone symbols outside of `some_func1` but there should be some operators to connect standalone symbols into a single expression. 

In other words, if we ever call `insideFunction` it means top level was Function, so `symbol instanceof Function would be a Function I referred as `some_func2` above.

Another thought: The whole logic of this validator is "do not throw inside Function" and fix is "do not throw when have nested functions", so I think we can even remove visitFunction completely? Can you please try it and if some test fails, point it out?